aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -941,6 +949,71 @@ public void run() {
         }, "Mqtt Connect Thread").start();
     }
 
+    /**
+     * Initializes the MQTT session and connects to the specified MQTT server
+     * using custom authorization tokens.
+     *
+     * @param tokenKeyName This specifies the key name that your device chooses, which indicates the
+     *                     token in the custom authorization HTTP request header. It is a string;
+     *                     the minimum length is 1 character, and the maximum length is 128 characters.
+     * @param token This specifies the custom authorization token to authorize the request to the AWS
+     *              IoT gateway. It is a string; the minimum length is 1 character, and the maximum
+     *              length is 1,024 characters.
+     * @param tokenSignature This is the token signature generated by using the private key with the
+     *                       SHA256withRSA algorithm.
+     * @param customAuthorizer Name of the AWS IoT custom authorizer.
+     * @param statusCallback When a new MQTT session status is received, a function of callback will
+     *                       be called with new connection status.
+     */
+    public void connect(
+            final String tokenKeyName,
+            final String token,
+            final String tokenSignature,
+            final String customAuthorizer,
+            final AWSIotMqttClientStatusCallback statusCallback) {
+
+        if (tokenKeyName == null || token == null || tokenSignature == null) {
+            throw new IllegalArgumentException("tokenKeyName/token/tokenSignature cannot be null");
+        }
+
+        this.userStatusCallback = statusCallback;
+
+        // Do nothing if Connecting, Connected or Reconnecting
+        if (connectionState != MqttManagerConnectionState.Disconnected) {
```

Comment:
Change this to `NonNull.equals(possibleNull)`. See if there are any more of these in this PR, remaining.

Reply:
> Are the SDK unit tests supposed to be passing, at this point?

Yep. I suppose there was a transient failure in the last circleci build. It all passes in the latest build.

Reply:
I came across an interesting SO post on whether to use the operator `==` or `equals()` method for enum comparison and it looks like the [operator is preferred](https://stackoverflow.com/questions/1750435/comparing-java-enum-members-or-equals) because of null-safety and other reasons but they are the same for all practical purposes. I have updated the occurrences in PR with `NonNull.equals(possibleNull)` pattern.